### PR TITLE
create VLAN based network

### LINKED ISSUE #327: OSP install with VLAN tenant networks
We need OSP install with vlan provider networks to workaround the double encap problem when not using kuryr. We also need to test OCP install on top of this cluster.

### PR DESCRIPTION
**Pull request by masco**
to create a VLAN based tenant network set vlan_provider_network is true and
set the below config in extra heat config
- NeutronBridgeMappings=openshift:br-openshift,datacentre:br-ex
- NeutronNetworkVLANRanges=openshift:500:1000,datacentre:1:500

fixes #327

### CODE REVIEW

**Review by smalleni — COMMENTED** [review]
@venkataanil @masco Do we not need a new bridge mapping  for he openshift_vlan_interface?

**Comment by venkataanil** [comment]
> @venkataanil @masco Do we not need a new bridge mapping for he openshift_vlan_interface?

Yes

**Comment by venkataanil** [comment]
@masco your PR is incomplete. It is not defining needed THT params i.e

NeutronNetworkType: vlan
network_vlan_ranges=openshift:500:1000,datacentre:1:500

File: templates/compute.yaml.j2 [hunk 1]
```diff
@@ -293,6 +293,17 @@ resources:
                               -
                                 ip_netmask:
                                     get_param: TenantIpSubnet
```
Comment (venkataanil) [inline templates/compute.yaml.j2:295]: {% endif %} is missing?

File: templates/compute.yaml.j2 [hunk 2]
```diff
@@ -274,7 +274,7 @@ resources:
                               -
                                 ip_netmask:
                                     get_param: StorageIpSubnet
```
Comment (venkataanil) [inline templates/compute.yaml.j2:276]: Add TenantIpSubnet here in case of vlan network

File: templates/compute.yaml.j2 (hunk 1 above)
Comment (masco) [reply templates/compute.yaml.j2:295]: it is present at line number 287

File: templates/compute.yaml.j2 (hunk 2 above)
Comment (masco) [reply templates/compute.yaml.j2:276]: if we set vlan_provider_network = true, it will be added here, since the below condition fails

**Comment by smalleni** [comment]
@masco @venkataanil Is this working now and is the PR final or, are more changes expected?

**Comment by masco** [comment]
> @masco @venkataanil Is this working now and is the PR final or, are more changes expected?

@smalleni we are facing some issues while creating an instance. working on it and will update the PR after fixing the issue.

**Pull request merged**